tests/jballerina-unit-test/src/test/resources/test-src/query/order-by-clause.bal

```diff
@@ -502,3 +502,98 @@ function incrementCount(int i) returns int {
     int count = i + 2;
     return count;
 }
+
+function testQueryExprWithOrderByClauseWithArrayKey() {
+    record {|int[] t; string s;|}[] data1 = [
```

Comment:
Lets add few more tests with `record` type and `enum`

Reply:
It is not possible to use a record type as a key in the order by clause since it is not a ordered-type. https://ballerina.io/spec/lang/master/#ordering